include/eve/module/real/core/function/promoted/simd/x86/mul.hpp

```diff
@@ -0,0 +1,243 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Contributors & Maintainers
+  SPDX-License-Identifier: MIT
+*/
+//==================================================================================================
+#pragma once
+
+#include <eve/detail/abi.hpp>
+#include <eve/forward.hpp>
+#include <type_traits>
+#include <eve/concept/compatible.hpp>
+#include <eve/concept/value.hpp>
+#include <eve/detail/implementation.hpp>
+#include <eve/detail/function/operators.hpp>
+#include <eve/traits/common_compatible.hpp>
+#include <eve/function/bit_xor.hpp>
+#include <eve/function/combine.hpp>
+#include <eve/function/converter.hpp>
+#include <eve/function/if_else.hpp>
+#include <eve/function/is_gtz.hpp>
+#include <eve/function/promoted.hpp>
+#include <eve/detail/apply_over.hpp>
+#include <eve/pattern.hpp>
+#include <concepts>
+#include <bit>
+
+namespace eve::detail
+{
+  template<unsigned_scalar_value T, typename N>
+  EVE_FORCEINLINE
+  auto mul_(EVE_SUPPORTS(sse2_), promoted_type const &, wide<T, N> a, wide<T, N> b) noexcept
```

Comment:
as the tests shows, the intrinsics used here are SSE4+, so maybe we can just `EVE_SUPPORTS(sse4_1)` ?